packages/expo/src/provider/ClerkProvider.tsx

```diff
@@ -1,30 +1,40 @@
-import './polyfills';
+import '../polyfills';
 
-import type { ClerkProviderProps as ClerkReactProviderProps } from '@clerk/clerk-react';
 import { ClerkProvider as ClerkReactProvider } from '@clerk/clerk-react';
-import React from 'react';
+import * as WebBrowser from 'expo-web-browser';
 
-import type { TokenCache } from './cache';
-import { isReactNative } from './runtime';
+import type { TokenCache } from '../cache/types';
+import { isNative, isWeb } from '../utils/runtime';
 import { getClerkInstance } from './singleton';
 
-export type ClerkProviderProps = ClerkReactProviderProps & {
+export type ClerkProviderProps = React.ComponentProps<typeof ClerkReactProvider> & {
   tokenCache?: TokenCache;
 };
 
+const SDK_METADATA = {
+  name: PACKAGE_NAME,
+  version: PACKAGE_VERSION,
+};
+
 export function ClerkProvider(props: ClerkProviderProps): JSX.Element {
   const { children, tokenCache, publishableKey, ...rest } = props;
   const pk = publishableKey || process.env.EXPO_PUBLIC_CLERK_PUBLISHABLE_KEY || process.env.CLERK_PUBLISHABLE_KEY || '';
 
+  if (isWeb()) {
+    // This is needed in order for useOAuth to work correctly on web.
+    WebBrowser.maybeCompleteAuthSession();
+  }
+
   return (
     <ClerkReactProvider
       // Force reset the state when the provided key changes, this ensures that the provider does not retain stale state.
       // See JS-598 for additional context.
       key={pk}
       {...rest}
       publishableKey={pk}
-      Clerk={getClerkInstance({ publishableKey: pk, tokenCache })}
-      standardBrowser={!isReactNative()}
+      sdkMetadata={SDK_METADATA}
+      Clerk={isNative() ? getClerkInstance({ publishableKey: pk, tokenCache }) : null}
+      standardBrowser={!isNative()}
```

Comment:
How does bundling work here ? 
`getClerkInstance` requires the `clerk-js` esm bundle, where as we actually only want to hotload on web. Does this mean we are loading clerk-js bundled and clerk-js hotload while on web ?

Or will metro drop `getClerkInstance` along side the import from clerk-js resulting in clerk-js only beeing hotloaded ?

Reply:
Metro will handle the bundling here it will leverage platform-specific extensions and will either load `singleton` which bundles `clerk-js` or `singleton.web` which does not bundle `clerk-js` and makes the hot loading available.